### PR TITLE
Clean up macros and fix bug in `shared_priority_queue_scheduler`

### PR DESCRIPTION
**Pull request by biddisco**
A small fix to the scheduler where a wrong param is set, plus some debug cleanup that is needed for other cuda/mpi polling branches

### CODE REVIEW

**Comment by biddisco** [comment]
bors try

File: libs/pika/schedulers/include/pika/schedulers/shared_priority_queue_scheduler.hpp [hunk 1]
```diff
@@ -783,8 +782,7 @@ namespace pika::threads {
             execution::thread_schedule_hint schedulehint, bool allow_fallback,
             execution::thread_priority priority = execution::thread_priority::normal) override
         {
-            schedule_work(thrd, schedulehint, allow_fallback, false,
-                priority = execution::thread_priority::normal);
```
Comment (msimberg) [inline libs/pika/schedulers/include/pika/schedulers/shared_priority_queue_scheduler.hpp:787]: No action required, just FYI: `nvc++` actually complained about this assignment, so this is also fixed on #606.

File: libs/pika/threading_base/src/print.cpp [hunk 2]
```diff
@@ -11,6 +11,12 @@
 #include <cstdint>
 #include <thread>
 
+#if defined(__linux) || defined(linux) || defined(__linux__)
+# include <linux/unistd.h>
+# include <sys/mman.h>
+# define DEBUGGING_PRINT_LINUX
```
Comment (msimberg) [inline libs/pika/threading_base/src/print.cpp:17]:
```suggestion
# define PIKA_DEBUGGING_PRINT_LINUX
```

**Comment by msimberg** [comment]
bors merge

**Pull request merged**